src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5619,6 +5619,14 @@ protected override void VisitCatchBlock(BoundCatchBlock node, ref LocalState fin
             base.VisitCatchBlock(node, ref finallyState);
         }
 
+        public override BoundNode VisitLockStatement(BoundLockStatement node)
+        {
+            VisitRvalue(node.Argument);
+            CheckPossibleNullReceiver(node.Argument);
```

Comment:
The `lock` statement is implemented by using the `Monitor.Enter` method. This change means we're picking the langauge definition of `lock`, cannot accept `null` values, instead of what the API of `Monitor.Enter` dictates. I'm fine with that, locking on `null` does seem kind of pointless, but wanted to make sure we're explicit about that.